### ISSUE TITLE
TracingRabbitListenerAdvice can't handle List of messages

### ISSUE THREAD

**Issue by savvadem** [openzipkin/brave#1240]
## Describe the Bug
TracingRabbitListenerAdvice fails in runtime, when I configure ``` SimpleRabbitListenerContainerFactory``` with batching.
## Steps to Reproduce
Create ``` SimpleRabbitListenerContainerFactory``` with following configuration
```
   public SimpleRabbitListenerContainerFactory batchRabbitListenerContainerFactory(
            ConnectionFactory cachingConnectionFactory) {
        SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
        factory.setConnectionFactory(cachingConnectionFactory);
        factory.setBatchSize(100);
        factory.setBatchListener(true);
        factory.setConsumerBatchEnabled(true);
        factory.setTransactionManager(rabbitTransactionManager(cachingConnectionFactory));
        return factory;
    }
```
create listener which consumes list of messages
```
    @RabbitListener(queues = "test.queue, containerFactory = "batchRabbitListenerContainerFactory")
    public void listener(@Payload List<Message> messages) {
        log.info("recieved batch {}", messages.size());
        for (Message message : messages) {
            log.info(message.toString());
        }
    }
```
when messages come container execute method ``` org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer#executeWithList``` with a list, and fails on ```brave.spring.rabbit.TracingRabbitListenerAdvice#invoke``` trying to cast ```List<Message>``` to ```Message```.
Stacktrace:
```
ava.lang.ClassCastException: java.util.ArrayList cannot be cast to org.springframework.amqp.core.Message
	at brave.spring.rabbit.TracingRabbitListenerAdvice.invoke(TracingRabbitListenerAdvice.java:75) ~[brave-instrumentation-spring-rabbit-5.10.1.jar:na]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.$Proxy204.invokeListener(Unknown Source) ~[na:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:1486) ~[spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:1481) ~[spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:1421) ~[spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.executeWithList(SimpleMessageListenerContainer.java:1023) [spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:1012) [spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.lambda$receiveAndExecute$2(SimpleMessageListenerContainer.java:897) [spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:140) ~[spring-tx-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:891) [spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$1600(SimpleMessageListenerContainer.java:81) [spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.mainLoop(SimpleMessageListenerContainer.java:1284) ~[spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1190) ~[spring-rabbit-2.2.5.RELEASE.jar:2.2.5.RELEASE]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_144]
```
## Expected Behaviour
no exception should be raised on batch.

current workaround is to disable via
``` spring.sleuth.messaging.enabled: false```

**Comment by savvadem** [comment]
Oh looks like there is already issue for this https://github.com/openzipkin/brave/issues/1236.
Didn't find it by google, so feel free to close as duplicate.

**Comment by adriancole** [comment]
this will be fixed in 5.12.4
